ledger/ledger-api-common/src/main/scala/com/digitalasset/platform/server/api/validation/FieldValidations.scala

```diff
@@ -156,12 +158,22 @@ class FieldValidations private (errorFactories: ErrorFactories) {
         if (duration.isNegative)
           Left(invalidField(fieldName, "Duration must be positive", definiteAnswer = Some(false)))
         else if (duration.compareTo(maxDeduplicationTime) > 0)
-          Left(invalidField(fieldName, exceedsMaxDurationMessage, definiteAnswer = Some(false)))
+          Left(
+            invalidField(
```

Comment:
I would keep this method simple and create a new one that dispatches your desired error code. e.g. `invalidDeduplicationPeriod`

Reply:
Then, it would help if you would add a small test to it to `ErrorFactoriesSpec` like we have for others.